### PR TITLE
Scalarlike now just goes by its name in equinox ecosystem documentation

### PR DESCRIPTION
**Pull request by johannahaffner**
As suggested in https://github.com/patrick-kidger/optimistix/pull/109. 

I built the optimistix documentation with jaxtyping from this branch, and it works splendidly.

### CODE REVIEW

**Comment by johannahaffner** [comment]
In optimistix, we use it to type values that should be `ArrayLike` with shape `()`. So that extra bit of information would get stripped along with the rest. I think it should be clear that e.g. a `learning_rate` is a single scalar value.

**Comment by patrick-kidger** [comment]
I agree, it should be clear! Which is why I'm thinking of stripping that information here for (a) readability for those unfamiliar with jaxtyping and (b) consistency.

As an alternative, we could consider turning on jaxtyping annotations in documentation for the whole ecosystem? The library has quite possibly achieved critical mass at this point, that a large enough fraction of readers will know what it means?

**Comment by johannahaffner** [comment]
> As an alternative, we could consider turning on jaxtyping annotations in documentation for the whole ecosystem? The library has quite possibly achieved critical mass at this point, that a large enough fraction of readers will know what it means?

I would prefer that, I find jaxtyping's expressiveness very useful! 

To get back to the previous example - for a learning rate it might not be too confusing to use `ArrayLike`, since we can assume that people know that this is a single scalar (small) value. But take `lambda_0` in `IndirectDampedNewtonDescent` - annotating this as `ArrayLike` could lead users to think that this could be a vector quantity ([here](https://github.com/patrick-kidger/optimistix/blob/0fcce03c3f2533e6df6cff9c9835e0110582cbdb/optimistix/_solver/levenberg_marquardt.py#L173)).

**Comment by patrick-kidger** [comment]
In that case at minimum let's replace this here to just return `Shaped[jaxtyping.ArrayLike, ""]` rather than `Shaped[jax.typing.ArrayLike, ""]` so that we redispatch to the doc-overloaded version of `ArrayLike` that we have! That'll then give us consistent behaviour regardless of the choice of using jaxtyping annotations in docs.

**Comment by patrick-kidger** [comment]
Haha, take a look at #297 to see what I mean :)

**Comment by johannahaffner** [comment]
Hahaha I see! 🙈
Thanks!
